### PR TITLE
Health Check Variables Customization

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Adds variables to disable component health checks
Adds user/pass variables for the health checks

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All scenarios should continue to function
Add basic auth scenario that proves the effectiveness of new variables

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.variables/defaults/main.yml [hunk 1]
```diff
@@ -949,3 +970,33 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+### User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
+kafka_rest_health_check_user: "{{kafka_rest_ldap_user}}"
+
+### Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
+kafka_rest_health_check_password: "{{kafka_rest_ldap_password}}"
+
+### User for authenticated Control Center Health Check. Set if using customized security like Basic Auth
+control_center_health_check_user: "{{control_center_ldap_user}}"
+
+### Password for authenticated Control Center Health Check. Set if using customized security like Basic Auth
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:1001]: run docs.py

File: VARIABLES.md [hunk 2]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
 
 ***
 
+### mds_health_check_user
+
+User for authenticated MDS Health Check. Only relevant if rbac_enabled: true. Likely leave as is
```
Comment (JumaX) [inline VARIABLES.md:1825]:
```suggestion
User for authenticated MDS Health Check. Only relevant if rbac_enabled: true.
```

File: VARIABLES.md [hunk 3]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{mds_super_user}}"
+
+***
+
+### mds_health_check_password
+
+Password for authenticated MDS Health Check. Only relevant if rbac_enabled: true. Likely leave as is
```
Comment (JumaX) [inline VARIABLES.md:1833]:
```suggestion
Password for authenticated MDS Health Check. Only relevant if rbac_enabled: true. 
```

File: VARIABLES.md [hunk 4]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{mds_super_user_password}}"
+
+***
+
+### kafka_broker_rest_health_check_user
+
+User for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1841]:
```suggestion
User for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Authd.
```

File: VARIABLES.md [hunk 5]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{mds_super_user}}"
+
+***
+
+### kafka_broker_rest_health_check_password
+
+Password for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1849]:
```suggestion
Password for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 6]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{mds_super_user_password}}"
+
+***
+
+### schema_registry_health_check_user
+
+User for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1857]:
```suggestion
User for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 7]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{schema_registry_ldap_user}}"
+
+***
+
+### schema_registry_health_check_password
+
+Password for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1865]:
```suggestion
Password for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 8]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{schema_registry_ldap_password}}"
+
+***
+
+### kafka_connect_health_check_user
+
+User for authenticated Connect Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1873]:
```suggestion
User for authenticated Connect Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 9]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{kafka_connect_ldap_user}}"
+
+***
+
+### kafka_connect_health_check_password
+
+Password for authenticated Connect Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1881]:
```suggestion
Password for authenticated Connect Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 10]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Default:  "{{kafka_connect_ldap_password}}"
+
+***
+
+### ksql_health_check_user
+
+User for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1889]:
```suggestion
User for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 11]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+User for authenticated Control Center Health Check. Set if using customized security like Basic Auth
+
+Default:  "{{control_center_ldap_user}}"
+
+***
+
+### control_center_health_check_password
+
+Password for authenticated Control Center Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1929]:
```suggestion
Password for authenticated Control Center Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 12]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
+
+Default:  "{{kafka_rest_ldap_password}}"
+
+***
+
+### control_center_health_check_user
+
+User for authenticated Control Center Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1921]:
```suggestion
User for authenticated Control Center Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 13]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
+
+Default:  "{{kafka_rest_ldap_user}}"
+
+***
+
+### kafka_rest_health_check_password
+
+Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1913]:
```suggestion
Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 14]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+Password for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
+
+Default:  "{{ksql_ldap_password}}"
+
+***
+
+### kafka_rest_health_check_user
+
+User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1905]:
```suggestion
User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth.
```

File: VARIABLES.md [hunk 15]
```diff
@@ -1764,6 +1820,118 @@ Default:  "{{control_center_telemetry_enabled}}"
...
+
+User for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
+
+Default:  "{{ksql_ldap_user}}"
+
+***
+
+### ksql_health_check_password
+
+Password for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline VARIABLES.md:1897]:
```suggestion
Password for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 16]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
 
 ### Boolean to send cp-ansible Telemetry Metrics from Control Center. Currently only sends cp-ansible version data
 control_center_telemetry_ansible_labels_enabled: "{{control_center_telemetry_enabled}}"
+
+### User for authenticated MDS Health Check. Only relevant if rbac_enabled: true. Likely leave as is
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:974]:
```suggestion
### User for authenticated MDS Health Check. Only relevant if rbac_enabled: true. Likely leave as is.
```

File: roles/confluent.variables/defaults/main.yml [hunk 17]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+mds_health_check_user: "{{mds_super_user}}"
+
+### Password for authenticated MDS Health Check. Only relevant if rbac_enabled: true. Likely leave as is
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:977]:
```suggestion
### Password for authenticated MDS Health Check. Only relevant if rbac_enabled: true.

File: roles/confluent.variables/defaults/main.yml [hunk 18]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+mds_health_check_password: "{{mds_super_user_password}}"
+
+### User for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:980]:
```suggestion
### User for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 19]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_broker_rest_health_check_user: "{{mds_super_user}}"
+
+### Password for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:983]:
```suggestion
### Password for authenticated Kafka Admin API Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 20]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_broker_rest_health_check_password: "{{mds_super_user_password}}"
+
+### User for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:986]:
```suggestion
### User for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 21]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+schema_registry_health_check_user: "{{schema_registry_ldap_user}}"
+
+### Password for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:989]:
```suggestion
### Password for authenticated Schema Registry Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 22]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+schema_registry_health_check_password: "{{schema_registry_ldap_password}}"
+
+### User for authenticated Connect Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:992]:
```suggestion
### User for authenticated Connect Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 23]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_connect_health_check_user: "{{kafka_connect_ldap_user}}"
+
+### Password for authenticated Connect Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:995]:
```suggestion
### Password for authenticated Connect Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 24]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_connect_health_check_password: "{{kafka_connect_ldap_password}}"
+
+### User for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:998]:
```suggestion
### User for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 25]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+ksql_health_check_user: "{{ksql_ldap_user}}"
+
+### Password for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:1001]:
```suggestion
### Password for authenticated ksqlDB Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 26]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+ksql_health_check_password: "{{ksql_ldap_password}}"
+
+### User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:1004]:
```suggestion
### User for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 27]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_rest_health_check_user: "{{kafka_rest_ldap_user}}"
+
+### Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:1007]:
```suggestion
### Password for authenticated Rest Proxy Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 28]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+kafka_rest_health_check_password: "{{kafka_rest_ldap_password}}"
+
+### User for authenticated Control Center Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:1010]:
```suggestion
### User for authenticated Control Center Health Check. Set if using customized security like Basic Auth.
```

File: roles/confluent.variables/defaults/main.yml [hunk 29]
```diff
@@ -949,3 +970,45 @@ control_center_telemetry_enabled: "{{telemetry_enabled}}"
...
+control_center_health_check_user: "{{control_center_ldap_user}}"
+
+### Password for authenticated Control Center Health Check. Set if using customized security like Basic Auth
```
Comment (JumaX) [inline roles/confluent.variables/defaults/main.yml:1013]:
```suggestion
### Password for authenticated Control Center Health Check. Set if using customized security like Basic Auth.
```

**Review by JumaX — CHANGES_REQUESTED** [review]
@domenicbove just some minor changes.  Great work here!

**Review by JumaX — APPROVED** [review]
LGTM.

**Pull request merged**